spec/queries/get_change_offer_options_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+
+RSpec.describe GetChangeOfferOptions do
+  include CourseOptionHelpers
+
+  let(:ratifying_provider) { create(:provider) }
+  let(:self_ratified_course) { create(:course, :open_on_apply, provider: ratifying_provider) }
+  let(:externally_ratified_course) { create(:course, :open_on_apply, accredited_provider: ratifying_provider) }
+  let(:provider_user) { create(:provider_user) }
+
+  def service(provider_user, course)
+    GetChangeOfferOptions.new(user: provider_user, current_course: course)
+  end
+
+  def set_user_permission(provider, make_decisions)
+    provider_user.provider_permissions << ProviderPermissions.new(
+      provider: provider,
+      make_decisions: make_decisions,
+    )
+  end
```

Comment:
Can we utilise the `provider` factory instead as all examples set `make_decisions` to true

```ruby
let(:ratifying_provider) { create(:provider, :with_user) }
let(:provider_user) { ratifying_provider.provider_users.first }

Reply:
You're right that we seem to always set `make_decisions` to true, so I'll make this the default and override it in the scenario that needs it.

However, beyond `make_decisions`, a big factor in these tests is which provider your user is associated with. Using a factory would only make sense if `let(:provider_user)` was re-defined in context blocks, which you've suggested, but I don't think is appropriate for these tests.

Reply:
I've renamed the method to `set_provider` to make it clear a big part of using this method is setting which provider the user is associated with (training vs. ratifying).

Reply:
It's now `set_provider_with_make_decisions`.